examples/semi_supervised/plot_semi_supervised_versus_svm_iris.py

```diff
@@ -0,0 +1,86 @@
+"""
+===============================================================================
+Decision boundary of semi-supervised classifiers versus SVM on the Iris dataset
+===============================================================================
+
+A comparison for the decision boundaries generated on the iris dataset
+by Label Spreading, Self-training and SVM.
+
+This example demonstrates that Label Spreading and Self-training can learn
+good boundaries even when small amounts of labeled data are available.
+
+Note that Self-training with 100% of the data is omitted as it is functionally
+identical to training the SVC on 100% of the data.
+
+"""
+print(__doc__)
+
+# Authors: Clay Woolam   <clay@woolam.org>
+#          Oliver Rausch <rauscho@ethz.ch>
+# License: BSD
+
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn import datasets
+from sklearn.svm import SVC
+from sklearn.semi_supervised import LabelSpreading
+from sklearn.semi_supervised import SelfTrainingClassifier
+
+
+iris = datasets.load_iris()
+
+X = iris.data[:, :2]
+y = iris.target
+
+# step size in the mesh
+h = .02
+
+rng = np.random.RandomState(0)
+y_rand = rng.rand(y.shape[0])
+y_30 = np.copy(y)
+y_30[y_rand < 0.3] = -1  # set random samples to be unlabeled
+y_50 = np.copy(y)
+y_50[y_rand < 0.5] = -1
+# we create an instance of SVM and fit out data. We do not scale our
+# data since we want to plot the support vectors
+ls30 = (LabelSpreading().fit(X, y_30), y_30, 'Label Spreading 30% data')
+ls50 = (LabelSpreading().fit(X, y_50), y_50, 'Label Spreading 50% data')
+ls100 = (LabelSpreading().fit(X, y), y, 'Label Spreading 100% data')
+
+# the base classifier for self-training is identical to the SVC
+base_classifier = SVC(kernel='rbf', gamma=.5, probability=True)
+st30 = (SelfTrainingClassifier(base_classifier).fit(X, y_30),
+        y_30, 'Self-training 30% data')
+st50 = (SelfTrainingClassifier(base_classifier).fit(X, y_50),
+        y_50, 'Self-training 50% data')
+
+rbf_svc = (SVC(kernel='rbf', gamma=.5).fit(X, y), y, 'SVC with rbf kernel')
```

Comment:
It would be nice to have one rbf_svc_30, fitted on `y_30`.

Can you also print the achieved accuracy of all the models on the full (or better test) dataset?

Reply:
Hmm, I'm not really sure how we could do that without making the layout messy.